pkg/inspector/image-inspector.go

```diff
@@ -69,28 +69,32 @@ func (i *defaultImageInspector) Inspect() error {
 		return fmt.Errorf("Unable to connect to docker daemon: %v\n", err)
 	}
 
-	if _, err := client.InspectImage(i.opts.Image); err != nil {
-		log.Printf("Pulling image %s", i.opts.Image)
-		imagePullOption := docker.PullImageOptions{Repository: i.opts.Image}
-
-		var imagePullAuths *docker.AuthConfigurations
-		var authCfgErr error
-		if imagePullAuths, authCfgErr = i.getAuthConfigs(); authCfgErr != nil {
-			return authCfgErr
-		}
+	imageMetaBefore, inspectErrBefore := client.InspectImage(i.opts.Image)
 
-		// Try all the possible auth's from the config file
-		var authErr error
-		for _, auth := range imagePullAuths.Configs {
-			if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
-				break
-			}
-		}
-		if authErr != nil {
-			return fmt.Errorf("Unable to pull docker image: %v\n", authErr)
+	log.Printf("Pulling image %s", i.opts.Image)
+	imagePullOption := docker.PullImageOptions{Repository: i.opts.Image}
+
+	var imagePullAuths *docker.AuthConfigurations
+	var authCfgErr error
+	if imagePullAuths, authCfgErr = i.getAuthConfigs(); authCfgErr != nil {
+		return authCfgErr
+	}
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	for _, auth := range imagePullAuths.Configs {
+		if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
```

Comment:
I can do that in another PR. I think that all the errors must be "Authentication Failed" (can't see any other option write now, as the authentication have already been read from the user input) but it also can't hurt to collect it and it might give the user a feedback about which authentications were actually collected and that they were tried and failed.